### DISCUSSION TITLE
Is NeuroDesk backward compatible?

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#133] [Q&A] [answered]
@aswinnarayanan Just wonder if NeuroDesk is backward compatible?
I installed on our HPC tag '0b80cb05f48191f758e93e12a16f1c750c74cfde' (28/10/2020). Can I now add to it new containers, or I need to upgrade it first?

**Comment by civier** [comment, marked as answer]
@aswinnarayanan  Also, if I need to upgrade, do I need to download all containers anew, or can leave the old ones? People are already working with the system, so I also want to make sure the new version of Neurodesk works fine before rolling it out. What is the best way to do it?

**Reply by stebo85** [reply]
Dear @civier,

Yes, we aim for full backwards compatibility - so the best way to upgrade is to download a later container. The application containers from the old image can stay and new application containers will be added. Because of this it should also be possible to back to an older version without any issues.

**Reply by aswinnarayanan** [reply]
Updating neurodesk will give you the latest container list and menus
Instructions to update neurodesk are available [here](https://github.com/NeuroDesk/neurodesk#to-update)
You may have to run `bash install.sh` as well if you're updating from last year's version to latest, to get the software categories working.

The old containers do not have to be deleted (unless you want to save space)
Neurodesk will only download container image if a copy does not already exist, so you can feel free to run [download-all-containers](https://github.com/NeuroDesk/neurodesk#to-download-all-containers)

**Comment by civier** [comment]
@aswinnarayanan Thanks for your quick response, Aswin.
Because it's on the HPC, I don't really care for the menus and such. Is there any reason to update Neurodesk then?

**Reply by aswinnarayanan** [reply]
For HPC, latest versions has the support for cvmfs, the latest apps.json and software support. 

Older versions might be missing bugfixes and newer software containers.
